src/Symfony/Component/Serializer/Tests/SerializerTest.php

```diff
@@ -1050,6 +1050,58 @@ public function testCollectDenormalizationErrorsWithConstructor()
 
         $this->assertSame($expected, $exceptionsAsArray);
     }
+
+    /** @requires PHP 8.0 */
+    public function testCollectDenormalizationErrorsWithConstructorAndTypeError()
+    {
+        $json = '{"bool": [true]}';
+
+        $classMetadataFactory = new ClassMetadataFactory(new AnnotationLoader(new AnnotationReader()));
+        $extractor = new PropertyInfoExtractor([], [new PhpDocExtractor(), new ReflectionExtractor()]);
+
+        $serializer = new Serializer(
+            [
+                new ObjectNormalizer($classMetadataFactory, null, null, $extractor, new ClassDiscriminatorFromClassMetadata($classMetadataFactory)),
+            ],
+            ['json' => new JsonEncoder()]
+        );
+
+        try {
+            $serializer->deserialize($json, Php80WithPromotedTypedConstructor::class, 'json', [
+                DenormalizerInterface::COLLECT_DENORMALIZATION_ERRORS => true,
+            ]);
+
+            $this->fail();
+        } catch (\Throwable $th) {
```

Comment:
Maybe could you just catch `PartialDenormalizationException` to make the test more readable?

Reply:
yes, you're right I have just copied this test and modified but it can be simplified. Will update it.